src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/util/IndexUtils.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.util
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementIndices
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.IndicesAdminClient
+import org.elasticsearch.cluster.ClusterState
+import org.elasticsearch.cluster.metadata.IndexMetaData
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentType
+
+class IndexUtils {
+    companion object {
+        @Suppress("ObjectPropertyNaming")
+        const val _META = "_meta"
+        const val SCHEMA_VERSION = "schema_version"
+        const val NO_SCHEMA_VERSION = 1L
+
+        var indexManagementSchemaVersion: Long
+            private set
+        var indexManagementHistoryVersion: Long
+            private set
+
+        init {
```

Comment:
A bit confused by this classes logic. It seems like when it's initialized it's already parsing the mappings from disk and setting the schema versions, but then also during initIndexStateManagementIndex we pass in the mappings again from disk to updateIndexMapping  and call the getSchemaVersion again by parsing the mappings to do the comparison?

Reply:
Yes these mapping normally will be identical to what has already parsed to set the schema version but this function to pass-in mapping helps for integration test. It helps to change mapping schema version.